bakery/codec.go

```diff
@@ -85,25 +107,25 @@ func encodeCaveatIdV0(
 		return nil, errgo.Notef(err, "cannot generate random number for nonce")
 	}
 	data := make([]byte, 0, 1+publicKeyPrefixLen+KeyLen+NonceLen+1+binary.MaxVarintLen64+len(rootKey)+len(condition)+box.Overhead)
-	data = append(data, 0) //version
+	data = append(data, byte(Version2)) //version
```

Comment:
you seem to make a big struct and then append to it. If you're doing an initial large make can't you put all these in the right index?